### DISCUSSION TITLE
Skywalking oap pods failing

### DISCUSSION THREAD

**Discussion by krishnaprakashthokala** [apache/skywalking#12056] [Q&A]
Hi Team, 

I have been facing issues for several days to resolve below issues.
 1. I have installed skywalking with elasticsearch and elasticsearch pod is up and running
 2. Oap pod is failing , getting connection refused error. I installed minikube on my vm and i have enough resources on my machine.
 3. please help me how to resolve these issues.
 
![image](https://github.com/apache/skywalking/assets/71684984/79b86a0c-05c8-4414-b523-c1bd0a5ff354)
<img width="854" alt="helm4" src="https://github.com/apache/skywalking/assets/71684984/6be7bd14-fb86-4a44-83b5-b38c8b4cb44d">
<img width="854" alt="helm3" src="https://github.com/apache/skywalking/assets/71684984/3fb651e3-42f5-43d8-b601-d4adafc9892f">
<img width="707" alt="helm2" src="https://github.com/apache/skywalking/assets/71684984/f3a3375f-2515-434d-bc7a-f0b8bb4baa15">
<img width="564" alt="helm1" src="https://github.com/apache/skywalking/assets/71684984/32654f63-8949-424a-b6d6-185316e29833">

<img width="943" alt="helm5_uilogs" src="https://github.com/apache/skywalking/assets/71684984/481b0f3b-be41-4da5-9387-03d5fd562660">

**Comment by wu-sheng** [comment]
The error shows, your OAP server is not up. You should check the logs. 

As mentioned in many discussions, you should consider using one of these two.
- Docker quick start, https://skywalking.apache.org/docs/main/next/en/setup/backend/backend-docker/
- Showcase, https://skywalking.apache.org/docs/skywalking-showcase/next/readme/

Helm requires you to have the k8s knowledge and diagnose the issues

**Reply by krishnaprakashthokala** [reply]
Thanks @wu-sheng . I tried with Showcase also seems it is required more resources . so I installed it but showcase OAP pod also failed. I have installed with default settings.

And I checked oap pod logs, i dont see any errors , 
![image](https://github.com/apache/skywalking/assets/71684984/f05bc149-397e-4cf2-84ff-7389893dfc41)

I wanted to execute skywalking in k8s cluster only. Because we have to give demo to client. i thought is issue with elasticsearch but it has no issues, it is up and running.  what are the dependencies are required to run OAP , i see one database is required so in my case elasticsearch is up and running. do we need to check whether elastic search is properly working or not , I mean any specific indexes required. I will attach current indexes available in Elasticsearch,
<img width="886" alt="helm6_elasticsearch" src="https://github.com/apache/skywalking/assets/71684984/12b0794b-17c1-4187-8e8a-cf4a2c033460">

**Reply by Superskyyy** [reply]
@krishnaprakashthokala please attach the oap log as a file to here, the screenshot didn't show any useful info

**Reply by Superskyyy** [reply]
elasticsearch has to be healthy before OAP can boot up sucessfully.

**Reply by krishnaprakashthokala** [reply]
@Superskyyy @wu-sheng  please find the attached pod logs - elasticsearch ,oapinit, ui and oap
[oap.txt](https://github.com/apache/skywalking/files/14779248/oap.txt)
[skywalking-ui.txt](https://github.com/apache/skywalking/files/14779249/skywalking-ui.txt)
[oap-init.txt](https://github.com/apache/skywalking/files/14779250/oap-init.txt)
[elasticsearch.txt](https://github.com/apache/skywalking/files/14779251/elasticsearch.txt)

**Reply by Superskyyy** [reply]
I'm not seeing any errors in the OAP log, if you sure that's all the log then it looks like to be a minikube networking issue.

**Reply by Superskyyy** [reply]
also changing the default timeout for startup probe to higher might help

**Reply by krishnaprakashthokala** [reply]
@Superskyyy  yes I copied all logs, just i am running only one replicas . And only one thing i changed for elasticsearch .  I changed InitialDelayseconds to 300 for elasticsearch readinessProbe because, it is getting failed . after setting that , elasticsearch is not getting failed. but OAP pod only one failing . I see errors in helm-ui logs also those are also related to OAP server related only.

If minikube networking issue then how can we diagnose those, I mean what could be the reasons. Because same errors i see in showcase pods also. I mean showcase(oap) pods also were giving errors

**Reply by Superskyyy** [reply]
I'm afraid that I cannot help to trouble shoot k8s related stuff because there could be too many reasons. have you tried to changed the startup probe settings for oap server so it doesn't attempt to kill OAP so eagerly? (i.e., this looks like the reason why minikube keeps killing oap and make it look like a crash loop)

**Reply by Superskyyy** [reply]
@krishnaprakashthokala 
```
          failureThreshold: 9
          periodSeconds: 10
```
this means 90 seconds for oap to listen to the 0.0.0.0 interface, if its too short then startup probe will fail for the pod and will kick oap to crash.

**Reply by Superskyyy** [reply]
https://github.com/apache/skywalking-helm/blob/917a6ff8bb20c2dca843df4317dcd3d9a337dff6/chart/skywalking/values.yaml#L68 check here, change this to a larger value.

**Reply by krishnaprakashthokala** [reply]
Thanks @Superskyyy. 

Now the OAP pod is up and running after changing the startup probe values. However, I still see a few errors in the logs. I have attached the complete logs in a file.
Although the pod is running, I am still encountering probe failures. Upon inspection of the logs, I noticed errors that need attention.
[oap-latest.txt](https://github.com/apache/skywalking/files/14784809/oap-latest.txt)
<img width="941" alt="helm7_oap" src="https://github.com/apache/skywalking/assets/71684984/1e9c73f3-d45a-48e5-9a1d-98faf11d38b9">

**Reply by krishnaprakashthokala** [reply]
@Superskyyy  i see these errors in logs.
addr=10.244.3.200:58392, laddr=10.244.3.206:12800][h2c://skywalking-skywalking-helm-oap-5d885fc956-7z8xs/internal/l7check#HEAD] Response: {startTime=2024-03-28T05:54:56.618Z(1711605296618680), length=0B, duration=8832ms(8832280887ns), totalDuration=8832ms(8832938858ns), cause=com.linecorp.armeria.common.stream.ClosedStreamException: received a RST_STREAM frame: CANCEL, headers=[:status=0]}
com.linecorp.armeria.common.stream.ClosedStreamException: received a RST_STREAM frame: CANCEL
        at com.linecorp.armeria.server.Http2RequestDecoder.onRstStreamRead(Http2RequestDecoder.java:396) [armeria-1.24.3.jar:?]
        at io.netty.handler.codec.http2.Http2FrameListenerDecorator.onRstStreamRead(Http2FrameListenerDecorator.java:59) [netty-codec-http2-4.1.86.Final.jar:4.1.86.Final]
        at io.netty.handler.codec.http2.DefaultHttp2

**Reply by Superskyyy** [reply]
Grpc received a RST_STREAM frame often means various things. From here it looks like client side grpc request to oap timeout because server isn't ready yet. So still a similar issue. 

You should also consider fix up the machine later because it looks like something is very slow.

**Reply by wu-sheng** [reply]
About local machine performance, if you are going to run showcase or agent+OAP+UI+DB(Elasticsearch), make sure you have 10U20G available for Kind/Minikube. Applications and Elasticsearch require resources.

**Reply by krishnaprakashthokala** [reply]
Thanks @wu-sheng  and @Superskyyy 
I am using  separate system for skywalking and decreased replicas to 1 , currently running oap -1 and elasticsearch -1 pod running.
I saw that enough memory also available for my system. please see attached screen.
<img width="901" alt="helm9_memory" src="https://github.com/apache/skywalking/assets/71684984/3482d3f6-6a5b-4b23-8bbc-eb0d18545097">
please provide me any suggestions to resolve these issues .

**Reply by Superskyyy** [reply]
So it's still start up probe failed? I'm a bit confused because you said you changed the delays for it. Was it delayed long enough?

**Reply by krishnaprakashthokala** [reply]
@Superskyyy yes still it is failing. i kept below params for startupprobe. I increased sequently and currenlty below configurations applied.
startupProbe:
      failureThreshold: 30
      periodSeconds: 40

**Reply by krishnaprakashthokala** [reply]
@Superskyyy  , now i dont see any errors in pods but when i access I am getting blank screen.
![image](https://github.com/apache/skywalking/assets/71684984/7d325002-d6c3-4573-b17e-d6cf3985886c)
